cmd/aries-agentd/startcmd/start.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package startcmd
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/defaults"
+	"github.com/hyperledger/aries-framework-go/pkg/restapi"
+	"github.com/spf13/cobra"
+	"net/http"
+	"strings"
+)
+
+const AgentHostFlagName = "api-host"
+const AgentHostFlagShorthand = "a"
+const AgentHostFlagUsage = "Host Name:Port"
+
+const AgentInboundHostFlagName = "inbound-host"
+const AgentInboundHostFlagShorthand = "i"
+const AgentInboundHostFlagUsage = "Inbound Host Name:Port"
+
+const AgentDBPathFlagName = "db-path"
+const AgentDBPathFlagShorthand = "d"
+const AgentDBPathFlagUsage = "Path to database"
+
+const MissingHostErrorMessage = "Unable to start aries agentd, host not provided"
+const MissingInboundHostErrorMessage = "Unable to start aries agentd, HTTP Inbound transport host not provided"
+
+var logger = log.New("aries-framework/agentd")
+
+type server interface {
+	ListenAndServe(host string, router *mux.Router) error
+}
+
+type RealServer struct {
+	server http.Server
+}
+
+func (s *RealServer) ListenAndServe(host string, router *mux.Router) error {
+	return http.ListenAndServe(host, router)
+}
+
+func Cmd(server server) *cobra.Command {
+	startCmd := &cobra.Command{
+		Use:   "start",
+		Short: "Start an agent",
+		Long:  `Start an Aries agent controller`,
+
+		Run: func(cmd *cobra.Command, args []string) {
+			host, err := cmd.Flags().GetString(AgentHostFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			inboundHost, err := cmd.Flags().GetString(AgentInboundHostFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			dbPath, err := cmd.Flags().GetString(AgentDBPathFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			err = startAgent(server, host, inboundHost, dbPath)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
```

Comment:
I can't seem to test these logger.Errorf(err.Error()) lines because the Cobra framework actually prevents this "run" func from executing if the incorrect arguments are passed in. So lines 59, 64, 69 and 74 will never execute.